runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -402,14 +401,13 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 	/*MMINLINE*/ void
 	memcpyFromArray(void *destData, J9IndexableObject *srcObject, I_32 elementIndex, I_32 elementCount)
 	{
-		J9ROMArrayClass *romArrayClass = (J9ROMArrayClass*)J9GC_J9OBJECT_CLAZZ(srcObject)->romClass;
-		UDATA elementScale = romArrayClass->arrayShape & 0x0000FFFF;
+		UDATA elementSize = J9ARRAYCLASS_GET_STRIDE(J9GC_J9OBJECT_CLAZZ(srcObject));
 		if (isInlineContiguousArraylet(srcObject)) {
 			// If the data is stored contiguously, then a simple copy is sufficient.
 			void* srcData = getDataPointerForContiguous(srcObject);
-			switch(elementScale)
+			switch(elementSize)
```

Comment:
@dmitripivkine I dont think this is correct, A valuetype containing a single object reference will be treated like an int/float